dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/datavalueset/DefaultDataValueSetService.java

```diff
@@ -1200,18 +1204,20 @@ else if ( strategy.isDelete() )
 
                     if ( !dryRun )
                     {
-                        dataValueBatchHandler.updateObject( internalValue );
-
-                        auditBatchHandler.addObject( auditValue );
-
                         if ( dataElement.isFileType() )
                         {
-                            FileResource fr = fileResourceService.getFileResource( internalValue.getValue() );
+                            DataValue actualDataValue = dataValueService.getDataValue( dataElement, period, internalValue.getSource(), categoryOptionCombo, internalValue.getAttributeOptionCombo() );
+
+                            FileResource fr = fileResourceService.getFileResource( actualDataValue.getValue() );
 
```

Comment:
do we need a NPE check for actualDataValue here ?

Reply:
I did think about it. What do you suggest if actualDataValue happens to be null?